Assets/Expansion2/CityBanners/citybannermanager.lua

```diff
@@ -4234,7 +4237,11 @@ function CQUI_RealHousingFromImprovements(pCity, PlayerID, pCityID)
           if not kPlot:IsImprovementPillaged() then
             local kImprovementData = GameInfo.Improvements[eImprovementType].Housing;
             if kImprovementData == 1 then    -- farms, pastures etc.
```

Comment:
I am not familiar with this code at all, but I am guessing from the comments that "type 1" means "basic game terrain improvements" or something similar, while "type 2" means "civilization-specific terrain improvements". if that is the case, I would extract this sacred knowledge :) into functions with corresponding names. 

also, is there a common place that both expansion1 and expansion2 code can access? I think we can move at least some of the duplicated code there. not all of it, I would rather have duplication than the wrong level of abstraction. but some can probably be moved.

Reply:
yeah, I was looking for the macros/constants that mapped to those values and did not have luck doing so.  any time we can use those to make the code more clear is a better time

for the record it also kills me there's no "continue" concept with lua.  this code with the nested ifs is just hard to read.  There's kind of a concept of goto, but I don't think this version of Lua that Civ6 uses is friendly with it.

but just the loop would read so much easier if we could do this kind of thing:

for _, plotId in pairs(kCityPlots) do
  if Map.GetPlotDistance > 3 then
    continue
  end

  if plot:IsImprovementPillaged() then
   continue
  end

  if improvement == STANDARD_IMPROVEMENT then
   -- do the farm math
  elseif improvement == SPECIALIZED_IMPROVEMENT then
   -- do the specialized math
  end
end -- for loop

It can be cleaned up a bit, putting the specialized math into a function (to avoid some of that repeated code) but I did not want to lump that in with these changes.  This is another file that if we can do the separation of base file and specialized CQUI file, it would certainly help CQUI be more robust and likely easier to fix when the game changes underneath

Reply:
agree, let's keep refactorings in separate  branches.